throw/src/test/java/android/BluetoothDeviceTest.kt

```diff
@@ -49,16 +45,19 @@ class BluetoothDeviceTest {
     fun `connectGattOrThrow throws result cause on failure`() {
         val context = mockk<Context>()
         val bluetoothDevice = mockk<BluetoothDevice>()
-        val cause = GattStatusFailure(OnConnectionStateChange(GATT_FAILURE, STATE_DISCONNECTED))
 
         mockkStatic("com.juul.able.android.BluetoothDeviceKt") {
-            coEvery { bluetoothDevice.connectGatt(any()) } returns Failure.Connection(cause)
+            coEvery {
+                bluetoothDevice.connectGatt(any())
+            } returns Failure.Connection(TestException())
 
-            assertFailsWith<GattStatusFailure> {
+            assertFailsWith<TestException> {
```

Comment:
I could be missing something here, but what's the point of testing that this function throws the exception you've mocked it to return? What logic is actually being tested here?

Reply:
> What logic is actually being tested here?

Admittedly, not a lot.

The little value the test does offer, is to confirm that purpose behind the `throw` module is being met. Basically, the `throw` module provides extension functions for all the `GattIo` methods that throw rather than returning their normal result. Allowing consumers of the library to change how they interact with failure cases.

In this particular case we're essentially just testing that we'll throw instead of returning a `ConnectGattResult` (as an alternate means of propagating failure). Which boils down to testing:

```kotlin
when (val result = connectGatt(context)) {
    is Success -> result.gatt
    is Failure -> throw result.cause
}
```

So ya, the test doesn't add much value at all, other than preventing regressions from code changes in the future.

It was challenging to write tests for the `throw` module, as it does such a simple task, the tests felt overly simple as well. ¯\\_(ツ)\_/¯

Reply:
> preventing regressions from code changes in the future.

That's a pretty solid reason for a test in my book. 👍